src/components/panel/panel.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * @ngdoc module
+ * @name material.components.panel
+ */
+angular
+    .module('material.components.panel', [
+      'material.core'
+    ])
+    .service('$mdPanel', MdPanelService);
+
+/**
+ * @ngdoc service
+ * @name $mdPanel
+ * @module material.components.panel
+ *
+ * @description
+ * `$mdPanel` is a general component that can be used for any panels on the screen,
+ * such as a tooltip, dialog, or custom popup.
+ *
+ * @usage
+ * <hljs lang="js">
+ * (function(angular, undefined) {
+ *   ‘use strict’;
+ *
+ *   angular
+ *       .module('demoApp', ['ngMaterial'])
+ *       .controller('PizzaBuilderController', PizzaBuilderController)
+ *       .controller('PizzaDialogController', DialogController);
+ *
+ *   function PizzaBuilderController($element, $mdPanel) {
+ *     var panelRef;
+ *     var toppings = [
+ *       'pepperoni',
+ *       'sausage',
+ *       'ham',
+ *       'peppers',
+ *       'onions',
+ *       'pineapple'
+ *     ];
+ *
+ *     function showPizzaPanel($event) {
+ *       var config = {
+ *         parent: angular.element(document.body),
+ *         controller: DialogController,
+ *         controllerAs: 'ctrl',
+ *         position: $element,
+ *         targetEvent: $event,
+ *         template:
+ *           '<div class="pizza-panel">' +
```

Comment:
It's okay to use a fake `templateUrl` instead of a whole template for the example.